createDataset.py

```diff
@@ -25,162 +151,74 @@ def getWhatsAppData():
-	return responseDictionary
+def getDiscordData():
+    personName = raw_input('Enter your full Discord name: ')
+    # Putting all the file names in a list
+    allFiles = []
+    # Edit these file and directory names if you have them saved somewhere else
+    for filename in os.listdir('DiscordChatLogs'):
+        if filename.endswith(".txt"): 
+            allFiles.append('DiscordChatLogs/' + filename)
+    responseDictionary = dict() # The key is the other person's message, and the value is my response
+    # Going through each file, and recording everyone's messages to me, and my
+                                # responses
+    for currentFile in allFiles:
+        with open(currentFile, 'r') as openedFile:
+            allLines = openedFile.readlines()
+        data = ''.join(allLines)
+        otherPersonsMessage, myMessage = "",""
+        response_sets = re.findall(r'\[.+\] (?!' + re.escape(personName) + r').+\n(.+)\n{2}(?:\[.+\] ' + re.escape(personName) + r'\n(.+)\n{2})', data)
+        for response_set in response_sets:
+            otherPersonsMessage = response_set[0]
+            myMessage = response_set[1]
+            responseDictionary[otherPersonsMessage] = cleanMessage(myMessage)
+            otherPersonsMessage, myMessage = "",""
+    return responseDictionary
 
 def cleanMessage(message):
-	# Remove new lines within message
-	cleanedMessage = message.replace('\n',' ').lower()
-	# Deal with some weird tokens
-	cleanedMessage = cleanedMessage.replace("\xc2\xa0", "")
-	# Remove punctuation
-	cleanedMessage = re.sub('([.,!?])','', cleanedMessage)
-	# Remove multiple spaces in message
-	cleanedMessage = re.sub(' +',' ', cleanedMessage)
-	return cleanedMessage
+    # Remove new lines within message
+    cleanedMessage = message.replace('\n',' ').lower()
+    # Deal with some weird tokens
+    cleanedMessage = cleanedMessage.replace("\xc2\xa0", "")
+    # Remove punctuation
+    cleanedMessage = re.sub('([.,!?])','', cleanedMessage)
+    # Remove multiple spaces in message
+    cleanedMessage = re.sub(' +',' ', cleanedMessage)
+    return cleanedMessage
 
 combinedDictionary = {}
 if (googleData == 'y'):
-	print 'Getting Google Hangout Data'
-	combinedDictionary.update(getGoogleHangoutsData())
+    print 'Getting Google Hangout Data'
+    combinedDictionary.update(getGoogleHangoutsData())
 if (fbData == 'y'):
-	print 'Getting Facebook Data'
-	combinedDictionary.update(getFacebookData())
+    print 'Getting Facebook Data'
+    combinedDictionary.update(getFacebookData())
 if (linkedInData == 'y'):
-	print 'Getting LinkedIn Data'
-	combinedDictionary.update(getLinkedInData())
+    print 'Getting LinkedIn Data'
+    combinedDictionary.update(getLinkedInData())
 if (whatsAppData == 'y'):
-        print 'Getting whatsApp Data'
-        combinedDictionary.update(getWhatsAppData())
+    print 'Getting whatsApp Data'
+    combinedDictionary.update(getWhatsAppData())
+if (discordData == 'y'):
+    print 'Getting Discord Data'
```

Comment:
Probably need parentheses around these prints? Or if you're saving the Python3 conversion changes for another PR, that's fine too.

Reply:
won't need parentheses in python 2 actually. This whole PR is still in python2, I'm working on porting to python 3 though.